core/domain/search_services.py

```diff
@@ -117,21 +152,31 @@ def get_search_rank_from_exp_summary(exp_summary):
     return max(rank, 0)
 
 
-def index_collection_summaries(collection_summaries):
+def index_collection_summaries(
+    collection_summaries: List[collection_domain.CollectionSummary]
+) -> None:
     """Adds the collections to the search index.
 
     Args:
         collection_summaries: list(CollectionSummaryModel). List of
             Collection Summary domain objects to be indexed.
     """
+    # The argument `documents` of add_documents_to_index() is annotated
+    # with List[Dict[str, Any]] because this method can accept any kind
+    # of dictionaries, but here we are providing a strict type
+    # `List[DomainSearchDict]` which causes a conflict in type assignment
+    # and due to this MyPy throws an error. So, to silent the error, we used
+    # ignore here.
     platform_search_services.add_documents_to_index([
```

Comment:
The only way we can use generic here is by using typevar with upper bound of `Mapping[str, Any]`, because typedDict is only allowed to be a subtype of `Mapping` when its key is represented as `Any`. 

I also tried for ` T = Tyevar('T', bound=Mapping[str, Union[str, in, List[str]])` but its not working. Can you please help!!

Reply:
As we discussed, it's better to refactor `add_documents_to_index`at a later stage as an object which can be initialized by passing the `TypedDict` as shown below

https://mypy.readthedocs.io/en/stable/generics.html#defining-generic-classes